lib/poller/client.ex

```diff
@@ -15,30 +15,42 @@ defmodule Poller.Client do
 
   def notify_next(%__MODULE__{current_list: []} = client) do
     processes =
-      ProcessRegistry
-      |> Registry.lookup(client.registry_key)
-      |> Enum.map(fn {pid, _} -> pid end)
+      client.registry_key
+      |> get_processes()
 
     notify_next(%__MODULE__{client | current_list: processes})
   end
 
   def notify_next(client) do
     [process | rest] = client.current_list
-    notify(process)
+
+    process
+    |> valid_process?(client.registry_key)
+    |> notify(process)
+
     %__MODULE__{client | current_list: rest}
   end
 
-  defp notify(pid) when is_pid(pid) do
-    case Process.alive?(pid) do
-      true ->
-        Process.send(pid, :call, [])
+  defp get_processes(registry_key) do
+    ProcessRegistry
+    |>Registry.lookup(registry_key)
+    |> Enum.map(fn {pid, _} -> pid end)
+  end
+
+  defp valid_process?(process, registry_key) do
+    alive = Process.alive?(process)
+
+    registered =
+      registry_key
+      |> get_processes()
+      |> Enum.any?(& (&1 == process))
 
-      _ ->
-        :ok
-    end
+    alive && registered
```

Comment:
What do you think of short-circuiting with a `with` statement so it'll return false if the process isn't alive? That way you can skip checking if it's a registered process.

```elixir
with true <- Process.alive?(process),
     processes <- get_processes(registry_key) do
  Enum.any?(processes, & &1 == process)
end
```

Reply:
Sounds good to me.  Thank you